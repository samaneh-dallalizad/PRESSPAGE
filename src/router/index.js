import { createRouter, createWebHistory } from 'vue-router'
import HomePage from './../components/HomePage/index.vue'
import Users from './../components/Users/index.vue'

const routes = [

    {
        path: '/',
        name: 'Home',
        component: HomePage
    },
    {
        path: '/users',
        name: 'users',
        component: Users,
    }
]
const router = createRouter({
    history: createWebHistory(),
    routes
})
export default router
