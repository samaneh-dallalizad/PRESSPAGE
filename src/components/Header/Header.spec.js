import Vue from 'vue';
import { mount } from '@vue/test-utils'
import {it} from 'vitest';
import HeaderComponent from './index.vue';
import router from "../../router"

describe('HeaderComponent', () => {
  it('header must always be the same', async () => {
    router.push("/")
    await router.isReady()
    const wrapper = mount(HeaderComponent, {
      global: {
        plugins: [router]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})