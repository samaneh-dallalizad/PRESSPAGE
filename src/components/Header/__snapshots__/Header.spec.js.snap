// Vitest Snapshot v1

exports[`HeaderComponent > header must always be the same 1`] = `
<section
  class="header"
  data-v-301e671b=""
>
  <div
    class="mx-auto w-1/4 my-10"
    data-v-301e671b=""
  >
    <a
      aria-current="page"
      class="router-link-active router-link-exact-active"
      data-v-301e671b=""
      href="/"
    >
      <img
        alt="presspage"
        data-v-301e671b=""
        src="src/assets/logo_presspage_white.png"
      />
    </a>
  </div>
</section>
`;
